src/main/java/org/mozilla/iot/webthing/Thing.java

```diff
@@ -89,11 +89,27 @@ public JSONObject asThingDescription() {
         JSONObject events = new JSONObject();
 
         this.availableActions.forEach((name, value) -> {
-            actions.put(name, value.getMetadata());
+            JSONObject metadata = value.getMetadata();
```

Comment:
This link object creation process seems gross enough to warrant putting in its own method that can be shared between properties, actions, and events